frontend/src/utils/create-async-process.ts

```diff
@@ -13,10 +13,15 @@ export default function createAsyncProcess<T extends (...args: any[]) => any>(
   const active: CreateAsyncProcessReturn<T>['active'] = ref(false);
 
   const run: CreateAsyncProcessReturn<T>['run'] = async (...args) => {
-    active.value = true;
-    const result = await fn(...args);
-    active.value = false;
-    return result;
+    try {
```

Comment:
Suggestion: catching an error can be cumbersome in terms of: who in the call hierarchy does it and when. And alternative approach would be to extend the state of an async process and add an error field. So you can check if the process has finished (not active anymore) and if there is an error. Might be beneficial and possibility allows to have more clean code. Please think about out. Maybe I'm completely off and have no clue what I'm talking about. In this case just resolve the comment thread silently. 😜

PS: if we get started with this, there are probably more improvements here. 😬

Reply:
I cannot say that I'm fully understanding this but in general, I have something against this `createAsyncProcess` that I'm not sure what is it exactly! maybe, in simple words, I feel it makes things over-complicated for my poor mind!  
Anyway, we can discuss all the options that make handling errors a good experience for both developers and users while working on issue #507